### PR TITLE
i18n common components: modals

### PR DESCRIPTION
**Pull request by rebeccaalpert**
Addressed the following modals/components: 
* About modal
* Column management modal
* Edit labels modal
* Edit Pod count modal
* Edit Parallelism modal
* Managed resource save modal
* Delete modal
* EmptyBox (used in a couple of modals)
* Check annotations modal
    * Needs secondary components done (will add as part of second PR with details page tabs since it shares components with Environments tab).
* Taints modal
* Edit pod selector
* Tolerations modal
* Expand PVC modal

It looks like PatternFly adds default aria-labels to many components. We may have to go through and add props so we can pass in translated ones. I haven't addressed this yet.

I skipped the following modals from the shared kebab for now since they're in a package:
* Clone PVC modal
* Restore PVC modal

I'll circle back and handle those later when I'm further along with the rest of the story.

This is a part of https://issues.redhat.com/projects/CONSOLE/issues/CONSOLE-2416.

### CODE REVIEW

File: frontend/public/components/about-modal.tsx [hunk 1]
```diff
@@ -116,8 +121,9 @@ export const AboutModal: React.FC<AboutModalProps> = (props) => {
     >
       {!customBranding && (
         <p>
-          OpenShift is Red Hat&apos;s container application platform that allows developers to
-          quickly develop, host, and scale applications in a cloud environment.
+          {t(
+            'modal~OpenShift is Red Hat&apos;s container application platform that allows developers to quickly develop, host, and scale applications in a cloud environment.',
```
Comment (spadgett) [inline frontend/public/components/about-modal.tsx:125]: You don't want to use the `&apos;` entity here.

```suggestion
            'modal~OpenShift is Red Hat\'s container application platform that allows developers to quickly develop, host, and scale applications in a cloud environment.',
```

File: frontend/public/components/factory/modal.tsx [hunk 2]
```diff
@@ -146,7 +148,7 @@ export const ModalSubmitFooter: React.SFC<ModalSubmitFooterProps> = ({
           data-test-id="modal-cancel-action"
           onClick={onCancelClick}
         >
-          {cancelText || 'Cancel'}
+          {cancelText || t('modal~Cancel')}
```
Comment (spadgett) [inline frontend/public/components/factory/modal.tsx:151]: `Cancel` should be in public

File: frontend/public/components/modals/column-management-modal.tsx [hunk 3]
```diff
@@ -151,8 +159,8 @@ export const ColumnManagementModal: React.FC<ColumnManagementModalProps> = ({
       <ModalSubmitFooter
         inProgress={false}
         cancel={cancel}
-        submitText="Save"
-        resetText="Restore Default Columns"
+        submitText={t('modal~Save')}
```
Comment (spadgett) [inline frontend/public/components/modals/column-management-modal.tsx:162]: `Save` should go in public

File: frontend/public/components/modals/delete-modal.tsx [hunk 4]
```diff
@@ -0,0 +1,158 @@
+import * as _ from 'lodash-es';
```
Comment (spadgett) [inline frontend/public/components/modals/delete-modal.tsx:1]: Thank you for converting to TypeScript!

File: frontend/public/components/modals/expand-pvc-modal.jsx [hunk 5]
```diff
@@ -1,87 +1,77 @@
...
     });
-  }
+  };
+
+  const { kind, resource } = props;
+
+  const dropdownUnits = {
+    Mi: t('modal~MiB'),
+    Gi: t('modal~GiB'),
+    Ti: t('modal~TiB'),
```
Comment (spadgett) [inline frontend/public/components/modals/expand-pvc-modal.jsx:43]: These are units that shouldn't be translated.

File: frontend/public/components/modals/taints-modal.tsx [hunk 6]
```diff
@@ -13,173 +14,149 @@ import {
...
-                    <input
-                      type="text"
-                      className="pf-c-form-control taint-modal__input"
-                      value={c.value}
-                      onChange={(e) => this._change(e, i, 'value')}
-                    />
+  const effects = {
+    NoSchedule: t('modal~NoSchedule'),
+    PreferNoSchedule: t('modal~PreferNoSchedule'),
+    NoExecute: t('modal~NoExecute'),
```
Comment (spadgett) [inline frontend/public/components/modals/taints-modal.tsx:62]: These are specific API values that I don't think we want to translate.

File: frontend/public/components/modals/taints-modal.tsx [hunk 7]
```diff
@@ -13,173 +14,149 @@ import {
...
+            ))}
+          </>
+        )}
+        <Button className="pf-m-link--align-left" onClick={addRow} type="button" variant="link">
+          <PlusCircleIcon data-test-id="pairs-list__add-icon" className="co-icon-space-r" />
+          {t('modal~Add more')}
+        </Button>
+      </ModalBody>
+      <ModalSubmitFooter
+        errorMessage={undefined}
```
Comment (spadgett) [inline frontend/public/components/modals/taints-modal.tsx:145]: I don't think `errorMessage` should always be `undefined`?

File: frontend/public/components/modals/tolerations-modal.tsx [hunk 8]
```diff
@@ -13,236 +14,211 @@ import {
...
   };
 
-  _submit = (e: React.FormEvent<EventTarget>) => {
+  const [tolerations, setTolerations] = React.useState(getTolerationsFromResource() || []);
+
+  const { t } = useTranslation();
+
+  const operators = {
+    Exists: t('modal~Exists'),
+    Equal: t('modal~Equal'),
```
Comment (spadgett) [inline frontend/public/components/modals/tolerations-modal.tsx:31]: These are specific API values I don't think we want to translate.

File: frontend/public/components/modals/tolerations-modal.tsx [hunk 9]
```diff
@@ -13,236 +14,211 @@ import {
...
+  };
+
+  const effects = {
+    '': t('modal~All Effects'),
+    NoSchedule: t('modal~NoSchedule'),
+    PreferNoSchedule: t('modal~PreferNoSchedule'),
+    NoExecute: t('modal~NoExecute'),
```
Comment (spadgett) [inline frontend/public/components/modals/tolerations-modal.tsx:38]: We shouldn't translate these IMO.

**Review by spadgett — COMMENTED** [review]
Thanks @rebeccaalpert !

**Comment by rebeccaalpert** [comment]
Addressed feedback; let me know if you have other comments!

**Comment by rebeccaalpert** [comment]
Resource Quota test is passing locally. Maybe the test issue is a flake? Trying again.

/test e2e-gcp-console

File: frontend/public/components/modals/expand-pvc-modal.jsx [hunk 10]
```diff
@@ -1,87 +1,78 @@
...
-    };
-    this._handleRequestSizeInputChange = this._handleRequestSizeInputChange.bind(this);
-    this._cancel = this.props.cancel.bind(this);
-    this._submit = this._submit.bind(this);
-  }
+const ExpandPVCModal = withHandlePromise((props) => {
+  const defaultSize = validate.split(getRequestedPVCSize(props.resource));
+  const [requestSizeValue, setRequestSizeValue] = React.useState(defaultSize[0] || '');
+  const [requestSizeUnit, setRequestSizeUnit] = React.useState(defaultSize[1] || 'Gi');
+  const [errorMessage] = React.useState();
```
Comment (spadgett) [inline frontend/public/components/modals/expand-pvc-modal.jsx:14]:
```suggestion
  const [errorMessage] = React.useState<string>();
```

File: frontend/public/components/modals/taints-modal.tsx [hunk 11]
```diff
@@ -13,173 +14,150 @@ import {
   ModalTitle,
 } from '../factory';
 
-class TaintsModal extends PromiseComponent<TaintsModalProps, TaintsModalState> {
-  readonly state: TaintsModalState;
+const TaintsModal = withHandlePromise((props: TaintsModalProps) => {
+  const [taints, setTaints] = React.useState(props.resource.spec.taints || []);
+  const [errorMessage] = React.useState();
```
Comment (spadgett) [inline frontend/public/components/modals/taints-modal.tsx:19]:
```suggestion
  const [errorMessage] = React.useState<string>();
```

File: frontend/public/components/modals/tolerations-modal.tsx [hunk 12]
```diff
@@ -13,236 +14,212 @@ import {
...
+  const [errorMessage] = React.useState();
+  const getTolerationsFromResource = (): Toleration[] => {
+    const { resource } = props;
+    return props.resourceKind.kind === 'Pod'
       ? resource.spec.tolerations
       : resource.spec.template.spec.tolerations;
   };
 
-  _submit = (e: React.FormEvent<EventTarget>) => {
+  const [tolerations, setTolerations] = React.useState(getTolerationsFromResource() || []);
```
Comment (spadgett) [inline frontend/public/components/modals/tolerations-modal.tsx:26]:
```suggestion
  const [tolerations, setTolerations] = React.useState<Toleration[]>(getTolerationsFromResource() || []);
```

**Review by spadgett — COMMENTED** [review]
LGTM, just some nits on TypeScript types

**Comment by spadgett** [comment]
/assign @yapei @ahardin-rh @sferich888 
for QE / docs / CEE review

This is ready for testing.

**Comment by ahardin-rh** [comment]
Ack from the docs side :+1:

**Comment by spadgett** [comment]
> Ack from the docs side 👍

/label docs-approved

**Comment by rebeccaalpert** [comment]
Rebased.

**Review by spadgett — APPROVED** [review]
/lgtm

**Comment by rebeccaalpert** [comment]
Rebased.

**Review by spadgett — APPROVED** [review]
/lgtm

**Review by spadgett — APPROVED** [review]
/lgtm

**Pull request merged**